app/services/foreman/plugin.rb

```diff
@@ -275,5 +275,10 @@ def apipie_ignored_controllers(controllers = nil)
     def register_custom_status(klass)
       HostStatus.status_registry.add(klass)
     end
+
+    # register a provision method
+    def provision_method(name, friendly_name)
+      Host::Managed.register_provision_method(name, friendly_name)
```

Comment:
In that case, prefer storing the registered provisioning inside this plugin object to ensure it stays in memory (in development mode Host::Managed may get reloaded) and then in Host::Managed retrieve and append the values from the registered plugins.